### PR TITLE
feat: autosave journal

### PR DESCRIPTION
**Pull request by deewakar-k**

### CODE REVIEW

**Review by madhav-relish — APPROVED** [review]

**Comment by sajdakabir** [comment]
not saving my journal
https://github.com/user-attachments/assets/0502e542-89e8-4d66-8b62-290f7e1248a0

**Comment by deewakar-k** [comment]
> not saving my journal
> 
> https://github.com/user-attachments/assets/0502e542-89e8-4d66-8b62-290f7e1248a0
> 
> 
> 
> 

are you updated? you should not see muted color i fixed that

File: apps/frontend/src/components/TodayTextArea.tsx [hunk 1]
```diff
@@ -38,11 +47,53 @@ export const TodayTextArea = ({ selectedDate }: JournalProps): JSX.Element => {
...
+
+  useEffect(() => {
+    if (hasUnsavedChanges && !isLoading) {
+      const debounceTimer = setTimeout(saveJournal, 2000);
+      return () => clearTimeout(debounceTimer);
+    }
+  }, [content, hasUnsavedChanges, isLoading]);
+
+  return (
+    <div className="text-foreground">
```
Comment (deewakar-k) [inline apps/frontend/src/components/TodayTextArea.tsx:95]: take a look @sajdakabir

**Comment by sajdakabir** [comment]
Yes, I did update the code although I'm in your branch. Let me check again.

**Comment by deewakar-k** [comment]
> Yes, I did update the code although I'm in your branch. Let me check again.

check your browser or something as your editor should be white in color your editor is muted which cannot be whole editor is set to white

**Review by sajdakabir — APPROVED** [review]

**Pull request merged**